languages/csharp/concept-exercises/pattern-matching/.docs/introduction.md

```diff
@@ -0,0 +1,26 @@
+﻿# Introduction
+
+Pattern matching in C# is very useful because
```

Comment:
I feel that the introduction is slightly verbose and perhaps could be condensed a bit. Consider the following:

> The use cases for pattern matching are innumerable,  
but the use-case that comes to mind first is for
building compilers and parsers.

I don't mind having the compilers reference, but it might be possibly to rephrase and shorten things a bit.